e2e/kmeans/kmeans.py

```diff
@@ -0,0 +1,58 @@
+# k-means step, given data, `X` and centroids
+# https://realpython.com/numpy-array-programming/#clustering-algorithms
+import numpy as np
+import torch
+torch.set_default_device("cpu")
+import torch._dynamo.config as cfg
+cfg.numpy_ndarray_as_tensor = True
+
+
+# np.linalg.norm replacement (2-norm only), https://github.com/pytorch/pytorch/issues/105269
+def norm(a, axis):
+    s = (a.conj() * a).real
```

Comment:
We don't support complex numbers in inductor, so this could very well be `a ** 2`

Reply:
so https://github.com/pytorch/pytorch/issues/105267 is a wontfix?

Reply:
yup

Reply:
https://github.com/Quansight-Labs/numpy_pytorch_interop/pull/165 has an example of working around not having complex dtypes.